nuvla/job_engine/job/actions/deployment_set_automatic_update.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+
+from ..actions import action
+
+import logging
+
+@action('deployment_set_automatic_update')
+class DeploymentSetAutomaticUpdateJob(object):
+
+    def __init__(self, job):
+        self.job = job
+        self.api = job.api
+        self.dep_set_id = self.job['target-resource']['href']
+
+    def _auto_update(self):
+        try:
+            dep_set = self.api.get(self.dep_set_id)
+            if dep_set.data['state'] in ['STARTED', 'UPDATED', 'PARTIALLY-STARTED', 'PARTIALLY-UPDATED']:
```

Comment:
I think to you can remove this check and look for the operation in operations list of the resource.

Reply:
Indeed, good idea.